pkg/core/resources/store/store.go

```diff
@@ -117,24 +116,31 @@ func (s *strictResourceStore) Close() error {
 	return nil
 }
 
-func ErrorResourceNotFound(rt model.ResourceType, name, mesh string) error {
-	return fmt.Errorf("Resource not found: type=%q name=%q mesh=%q", rt, name, mesh)
+type ResourceConflictError struct {
+	rType model.ResourceType
+	name  string
+	mesh  string
+	msg   string
 }
 
-func ErrorResourceAlreadyExists(rt model.ResourceType, name, mesh string) error {
-	return fmt.Errorf("Resource already exists: type=%q name=%q mesh=%q", rt, name, mesh)
+func (e *ResourceConflictError) Error() string {
+	return fmt.Sprintf("%s: type=%q name=%q mesh=%q", e.msg, e.rType, e.name, e.mesh)
 }
 
-func ErrorResourceConflict(rt model.ResourceType, name, mesh string) error {
-	return fmt.Errorf("Resource conflict: type=%q name=%q mesh=%q", rt, name, mesh)
+func (e *ResourceConflictError) Is(err error) bool {
+	return reflect.TypeOf(e) == reflect.TypeOf(err)
 }
 
-func IsResourceConflict(err error) bool {
-	return err != nil && strings.HasPrefix(err.Error(), "Resource conflict")
+func ErrorResourceAlreadyExists(rt model.ResourceType, name, mesh string) error {
+	return &ResourceConflictError{msg: "resource already exists", rType: rt, name: name, mesh: mesh}
 }
 
-func ErrorResourcePreconditionFailed(rt model.ResourceType, name, mesh string) error {
-	return fmt.Errorf("Resource precondition failed: type=%q name=%q mesh=%q", rt, name, mesh)
+func ErrorResourceConflict(rt model.ResourceType, name, mesh string) error {
+	return &ResourceConflictError{msg: "resource conflict", rType: rt, name: name, mesh: mesh}
+}
+
+func ErrorResourceNotFound(rt model.ResourceType, name, mesh string) error {
+	return fmt.Errorf("Resource not found: type=%q name=%q mesh=%q", rt, name, mesh)
 }
```

Comment:
```suggestion
type ResourceNotFoundError struct {
	rType model.ResourceType
	name  string
	mesh  string
}

func (e *ResourceNotFoundError) Error() string {
	return fmt.Sprintf("resource not found: type=%q name=%q mesh=%q", e.rType, e.name, e.mesh)
}

func (e *ResourceNotFoundError) Is(err error) bool {
	return reflect.TypeOf(err) == reflect.TypeOf(e)
}

func ErrorResourceNotFound(rt model.ResourceType, name, mesh string) error {
	return &ResourceNotFoundError{
		rType: rt,
		name:  name,
		mesh:  mesh,
	}
}
```

Reply:
This is out of scope of this PR. I agree we should do this at some point and we have: https://github.com/kumahq/kuma/issues/2604 This PR is already a good size I don't want to have it even bigger